packages/flutter_tools/test/general.shard/vmservice_test.dart

```diff
@@ -140,17 +135,27 @@ void main() {
       mockVMService,
     );
 
-    verify(mockVMService.registerService('flutterGetSkSL', 'Flutter Tools')).called(1);
-  }, overrides: <Type, Generator>{
-    Logger: () => BufferLogger.test()
+    expect(mockVMService.services, containsPair('flutterGetSkSL', 'Flutter Tools'));
+  });
+
+  testWithoutContext('VmService throws tool exit on service registration failure.', () async {
+    final MockVMService mockVMService = MockVMService()
+      ..errorOnRegisterService = true;
```

Comment:
I'm worried about my understanding of errors on Futures, and I'd feel a bit better if you wouldn't mind adding a test of registration failure when there's a flutterPrintStructuredErrorLogMethod as well.

Reply:
That doesn't use a service registration, it listens to an event stream. If that fails we keep on trucking since not getting those events isn't fatal